### PR TITLE
[NEXT RELEASE] Add new openshift-s2i-guide

### PR DESCRIPTION
**Pull request by vorburger**
see https://github.com/quarkusio/quarkus-quickstarts/pull/81
see https://github.com/quarkusio/quarkus/pull/1353

### CODE REVIEW

**Comment by vorburger** [comment]
@siamaksade @geoand @cescoffier @gastaldi

**Comment by geoand** [comment]
Maybe we should explain some of the non obvious points in the guide?
Like `quarkus.http.host` and `uberJar`?

File: _guides/openshift-s2i-guide.adoc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+In this section, we are going to leverage the S2I build mechanism of OpenShift.
+We use a Java S2I Builder, and therefore do not need a `Dockerfile` in this approach.
+You do not need to locally clone the Git repository, as it will be directly built since OpenShift.
+We are going to create an OpenShift `build` executing it:
+
+[source,shell]
+----
+# To build the image on OpenShift
+oc new-app fabric8/s2i-java:latest-java11~https://github.com/quarkusio/quarkus-quickstarts --context-dir=getting-started-openshift-s2i --name=quarkus-quickstart
```
Comment (gastaldi) [inline _guides/openshift-s2i-guide.adoc]: I am not sure if Fabric8 images should be promoted in the quickstarts. Is that the officially supported Java 11 S2I image?

**Comment by vorburger** [comment]
> Maybe we should explain some of the non obvious points in the guide?
> Like `quarkus.http.host` and `uberJar`?

@geoand I'm all for it... please feel free to follow up with more PRs on top of this work to improve it!

@gastaldi lemme try that other image (which ... there is whole other story.. anyway). Just FYI `_guides/openshift-s2i-guide.adoc` just moved from this PR over into https://github.com/quarkusio/quarkus/pull/1353, so I'll make the change over there and not here (after I tested it and if it works).

File: _guides/openshift-s2i-guide.adoc (hunk 1 above)
Comment (vorburger) [reply _guides/openshift-s2i-guide.adoc]: @gastaldi #1353 now documents using "that other" S2I builder image, instead.

**Comment by vorburger** [comment]
> > Maybe we should explain some of the non obvious points in the guide?
> > Like `quarkus.http.host` and `uberJar`?
> 
> @geoand I'm all for it... please feel free to follow up with more PRs on top of this work to improve it!

@geoand @gastaldi #1353 now has a Background chapter with some blurb about what's going on.

**Comment by vorburger** [comment]
Now that both https://github.com/quarkusio/quarkus-quickstarts/pull/81 as well as https://github.com/quarkusio/quarkus/pull/1353 have been merged, this is ready to go in (I'll Resolve the Conflict ASAP).

**Comment by vorburger** [comment]
Strictly speaking, https://github.com/quarkusio/quarkus/pull/1353/files only added doc for Java not Native S2I Building, but the README from https://github.com/quarkusio/quarkus/pull/1358/files has that, and https://github.com/quarkusio/quarkus/pull/1801 will bring it to https://quarkus.io, so to "fit in" the menu as-is now, I think this header makes sense.

@tqvarnst @siamaksade @cescoffier @gastaldi @geoand OK for you?

@dandreadis @dneary FYI

File: _guides/guides.md [hunk 2]
```diff
@@ -43,6 +43,7 @@ permalink: /guides/
 
 ### Cloud
 
+- [Deploying Native Applications on OpenShift with S2I]({{site.baseurl}}/guides/openshift-s2i-guide)
```
Comment (cescoffier) [inline _guides/guides.md]: Can you move this item as the third item of the list?

Comment (vorburger) [reply _guides/guides.md]: @cescoffier sure, done.

**Comment by cescoffier** [comment]
My mistake, this would need to be re-applied after the release.

**Pull request merged**